json-schema/schema.json

```diff
@@ -298,6 +316,52 @@
         }
       }
     },
+    "variable": {
+      "title": "Variable Object",
+      "type": "object",
+      "required": [
+        "dimensions"
+      ],
+      "properties": {
+        "type": {
+          "$ref": "#/definitions/variable_type"
```

Comment:
I think you can just directly put the content of the references entity here, because it's only used once.